openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/common.py

```diff
@@ -266,15 +253,45 @@ def build_dag_configs(ingestion_pipeline: IngestionPipeline) -> dict:
     }
 
 
-def send_failed_status_callback(workflow_config: OpenMetadataWorkflowConfig, context):
+def send_failed_status_callback(workflow_config: OpenMetadataWorkflowConfig, _):
     """
     Airflow on_failure_callback to update workflow status if something unexpected
     happens or if the DAG is externally killed.
+
+    We don't want to initialize the full workflow as it might be failing
+    on the `__init__` call as well. We'll manually prepare the status sending
+    logic.
+
+    In this callback we just care about:
+    - instantiating the ometa client
+    - getting the IngestionPipeline FQN
+    - if exists, update with `Failed` status
+
+    Here the workflow_config is already properly shaped, otherwise
+    the DAG deployment would fail.
+
+    More info on context variables here
+    https://airflow.apache.org/docs/apache-airflow/stable/templates-ref.html#templates-variables
     """
     logger.info("Sending failed status from callback...")
-    config = json.loads(workflow_config.json(encoder=show_secrets_encoder))
-    workflow = Workflow.create(config)
-    workflow.set_ingestion_pipeline_status(PipelineState.failed)
+
+    metadata_config = workflow_config.workflowConfig.openMetadataServerConfig
+    metadata = OpenMetadata(config=metadata_config)
+
+    if workflow_config.ingestionPipelineFQN:
```

Comment:
Should we log something if `workflow_config.ingestionPipelineFQN` is `None`, so people understand why it might not get updated?

Reply:
thanks @TeddyCr that makes sense. Added it